pages/api_differences.md.erb

```diff
@@ -0,0 +1,46 @@
+# API Differences Between REST and GraphQL
+
+Buildkite supports the use of both [REST](https://buildkite.com/docs/apis/rest-api) and [GraphQL](/docs/apis/graphql-api), but there is no full parity in the API between the two. Certain actions can only be performed using the REST API, but not GraphQL, and vice versa.
+
+{:toc}
+
+If you are only deciding which approach to take, you could consider GraphQL as your default answer to "how do I query a possibly-complicatedly-constrained subset of data?", and REST should be most helpful in a scenario of creating and modifying records.
+
+On this page we've collected the known limitation where some API features are only available with either REST or GraphQL.
+
+## Exclusive REST features
+
+* Buildkite's terraform provider uses GraphQL where it can, but it falls back to using REST for setting [`branch_configuration` and `github provider configuration`](/docs/pipelines/branch-configuration#step-level-branch-filtering) on pipelines.
```

Comment:
```suggestion
* Setting [`branch_configuration` and `github provider configuration`](/docs/pipelines/branch-configuration#step-level-branch-filtering) on pipelines.
```

Reply:
this feels more consistent and findable. If the Terraform is important we can add it in after the main sentence?

Reply:
These links should be to the relevant API section, not to the generic docs. If it isn't clear what those are we'll need to do a bit more work.